BAZProject/Commons/Commons.swift

```diff
@@ -7,64 +7,53 @@
-    case Upcoming = "Upcoming"
-    case Search =  "Search"
+/// enumerador para menejar de los tipos y sus urls
+enum CategoriesFilter: String {
+    case trending = "Trending"
+    case nowPlaying = "Now Playing"
+    case popular = "Popular"
+    case topRated = "Top Rated"
+    case upcoming = "Upcoming"
+    case search =  "Search"
+    case recommendation = "Recommendation Movies"
+    case similar = "similar Movies"
+    case cast = "cast"
     
     var url: String {
+        let baseURL = "https://api.themoviedb.org/3/movie/"
         switch self {
-        case .Trending:
+        case .trending:
             return "https://api.themoviedb.org/3/trending/movie/day?api_key="
-        case .NowPlaying:
-            return "https://api.themoviedb.org/3/movie/now_playing?api_key="
-        case .Popular:
-            return "https://api.themoviedb.org/3/movie/popular?api_key="
-        case .TopRated:
-            return "https://api.themoviedb.org/3/movie/top_rated?api_key="
-        case .Upcoming:
-            return "https://api.themoviedb.org/3/movie/upcoming?api_key="
-        case .Search:
+        case .nowPlaying:
+            return "\(baseURL)now_playing?api_key="
+        case .popular:
+            return "\(baseURL)popular?api_key="
+        case .topRated:
+            return "\(baseURL)top_rated?api_key="
+        case .upcoming:
+            return "\(baseURL)upcoming?api_key="
+        case .search:
             return "https://api.themoviedb.org/3/search/movie?api_key="
+        case .recommendation, .similar, .cast:
+            return baseURL
         }
     }
 }
 
-//MARK: - Extensiones
-
-// Metodos extras para el uso de peliculas con Strings
-extension String {
-    /// Formato del nombre de la pelicula para poder buscarla
-    func formatterMovieName() -> String {
-        self.replacingOccurrences(of: " ", with: "%20", options: NSString.CompareOptions.literal, range: nil) //Dividir palabras para la busqueda
-        
-    }
+//enum para UserDefauls
+enum UserDKeys: String {
+    case favorites = "favoritesMovies"
+    case favoriteTests = "favoriteTests"
 }
 
-// Metodos extras para la colocacion de las imagenes
-extension UIImageView{
-    /// Consulta desde una url para colocar la imagen
-    func loadImage(url: URL) -> URLSessionDownloadTask {
-        let session = URLSession.shared
-        let downloadTask = session.downloadTask(with: url){[weak self] url, response, error in
-            guard let self = self else { return }
-            if error == nil, let url = url, let data = try? Data(contentsOf: url), let image = UIImage(data: data){
-                DispatchQueue.main.async {
-                    self.image = image
-                }
-            }
-        }
-        downloadTask.resume()
-        return downloadTask
-    }
+//enum para Manejo de titulos
+enum StringsTitles: String {
+    case favorites = "Favoritos"
+    case search = "buscar"
+    case treding = "Trending"
 }
```

Comment:
```suggestion
/// Enum para Manejo de títulos.
enum Title: String {
    static let favorites: String  = "Favoritos"
    static let search: String = "buscar"
    static let treding: String = "Trending"
}
```
En estos casos, para usar los valores de una propiedad y no el iterar entre un enum lo mejor es declarar propiedades estáticas. Así sólo es necesario escribir esto: `Title.favorites` por ejemplo. Así evitas usar el `.rawValue`. Puedes aplicar lo mismo al enum de UserDKeys, que por cierto, podrías llamar `DefaultsKey`